aea/mail/base.py

```diff
@@ -368,7 +320,12 @@ def __init__(
 
         self._protocol_specification_id: PublicId = protocol_specification_id
         self._message = message
-        self._context = context if context is not None else EnvelopeContext()
+        if self.is_component_to_component_message:
+            enforce(
+                context is None,
+                "EnvelopeContext must be None for component to component messages.",
```

Comment:
Component to component messages specify the target and source explicitly. Hence, there is no need for a context.